### PR TITLE
[Merged by Bors] - feat(GroupTheory/SpecificGroups/Cyclic): unique_zpow_zmod

### PR DESCRIPTION
**Pull request by kbuzzard**
Suggested in #6342 .

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean [hunk 1]
```diff
@@ -347,6 +347,28 @@ theorem IsCyclic.image_range_card (ha : ∀ x : α, x ∈ zpowers a) :
...
+  refine ⟨n, (?_ : a ^ n = _), ?_⟩
+  · rw [← zpow_coe_nat, zpow_eq_zpow_iff_modEq, orderOf_eq_card_of_forall_mem_zpowers ha,
+      Int.modEq_comm, Int.modEq_iff_add_fac, ← ZMod.int_coe_zmod_eq_iff]
+  · intro y hy
+    rw [← zpow_coe_nat, zpow_eq_zpow_iff_modEq, orderOf_eq_card_of_forall_mem_zpowers ha,
+      ← ZMod.int_cast_eq_int_cast_iff] at hy
+    simp [hy]
+
+@[to_additive]
+lemma IsCyclic.ext {G : Type _} [Group G] [Fintype G] [IsCyclic G] {d : ℕ} {a b : ZMod d}
```
Comment (kbuzzard) [inline Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:363]:
```suggestion
lemma IsCyclic.ext {G : Type*} [Group G] [Fintype G] [IsCyclic G] {d : ℕ} {a b : ZMod d}
```

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean [hunk 2]
```diff
@@ -347,6 +347,28 @@ theorem IsCyclic.image_range_card (ha : ∀ x : α, x ∈ zpowers a) :
 #align is_cyclic.image_range_card IsCyclic.image_range_card
 #align is_add_cyclic.image_range_card IsAddCyclic.image_range_card
 
+@[to_additive]
+theorem IsCyclic.unique_zpow_zmod (ha : ∀ x : α, x ∈ zpowers a) (x : α) :
```
Comment (riccardobrasca) [inline Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:351]: LGTM, but maybe you can golf the proof using `zmodCyclicMulEquiv` (this results looks like a special case).

Thanks!

bors d+

Comment (kbuzzard) [reply Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:351]: @riccardobrasca I can't find `​zmodCyclicMulEquiv` in the docs or with docs search or with moogle. Can you give me more clues?

Comment (riccardobrasca) [reply Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:351]: [zmodCyclicMulEquiv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/SpecificGroups/Cyclic.html#zmodCyclicMulEquiv) does work?

Comment (kbuzzard) [reply Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:351]: ?! I must have been on the mathlib3 docs? Sorry for wasting your time!

Comment (riccardobrasca) [reply Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:351]: This is weird!

Comment (kbuzzard) [reply Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:351]: Getting back to the main point, I golfed my proof a little more and I can't see how to make it shorter with `zmodCyclicMulEquiv`; can you?

Comment (riccardobrasca) [reply Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:351]: I didn't have anything precise in mind, it only looks like a reasonable strategy. It's not very important.

**Pull request closed without merging**